trie/child_storage.go

```diff
@@ -0,0 +1,96 @@
+package trie
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/common"
+)
+
+// ChildStorageKeyPrefix is the prefix for all child storage keys
+var ChildStorageKeyPrefix = []byte(":child_storage:")
+
+// PutChild inserts a child trie into the main trie at key :child_storage:[keyToChild]
+func (t *Trie) PutChild(keyToChild []byte, child *Trie) error {
+	childHash, err := child.Hash()
+	if err != nil {
+		return err
+	}
+
+	key := append(ChildStorageKeyPrefix, keyToChild[:]...)
+	exists, err := t.Get(key)
+	if err != nil {
+		return err
+	}
+
+	if exists != nil {
+		err = t.Delete(key)
```

Comment:
You're sure we shouid be deleting this?

Reply:
yeah, whenever we write to the child storage the hash is gonna get updated. so we need to delete+re-insert 

however it might be nicer to have a trie update function so we don't need to have 2 lookups... I could add that